### PR TITLE
Fix AMM display

### PR DESCRIPTION
**Pull request by ThibautBremand**

### CODE REVIEW

File: packages/extension/src/utils/trustlines.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
+ * AMM Related
+ */
+export const LP_TOKEN = 'LP Token';
```
Comment (FlorianBouron) [inline packages/extension/src/utils/trustlines.ts:4]: Maybe:
```suggestion
export const LP_TOKEN_NAME = 'LP Token';
```

File: packages/extension/src/components/molecules/TokenDisplay/TokenInfo/TokenInfo.tsx [hunk 2]
```diff
@@ -25,17 +27,33 @@ export const TokenInfo = forwardRef((props: TokenInfoProps, ref: Ref<HTMLDivElem
...
+  }, [token]);
+
+  const displayToken = useMemo(() => {
+    return token.length > MAX_TOKEN_LENGTH && !isLPToken
+      ? `${token.slice(0, MAX_TOKEN_LENGTH)}...`
+      : token;
+  }, [token, isLPToken]);
+
+  const formattedIssuerAddress = useMemo(() => {
+    return issuerAddress && issuerAddress.length > 20
```
Comment (FlorianBouron) [inline packages/extension/src/components/molecules/TokenDisplay/TokenInfo/TokenInfo.tsx:45]: Is this possible that the address don't have the same size?

Comment (ThibautBremand) [reply packages/extension/src/components/molecules/TokenDisplay/TokenInfo/TokenInfo.tsx:45]: yes

File: packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx [hunk 3]
```diff
@@ -115,14 +144,51 @@ export const XRPLTransaction: FC<XRPLTxProps> = ({
...
   };
 
+  const renderCurrency = (params: { title: string; value: Currency }) => {
+    const { title, value } = params;
+    const currency =
+      value.currency.length === 40 ? convertHexCurrencyString(value.currency) : value.currency;
+    const formatted = value.issuer ? `${currency}\n${value.issuer}` : currency;
+    const hasTooltip = !!value.issuer;
+    return (
+      <>
```
Comment (FlorianBouron) [inline packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx:154]: You don't need to wrap in `<>`

File: packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx [hunk 4]
```diff
@@ -115,14 +144,51 @@ export const XRPLTransaction: FC<XRPLTxProps> = ({
     );
   };
 
+  const renderCurrency = (params: { title: string; value: Currency }) => {
+    const { title, value } = params;
+    const currency =
+      value.currency.length === 40 ? convertHexCurrencyString(value.currency) : value.currency;
```
Comment (FlorianBouron) [inline packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx:150]: You don't need to do this checking, this check is done within: `convertHexCurrencyString`.
const currency = convertHexCurrencyString(value.currency); should be enough

File: packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx [hunk 5]
```diff
@@ -115,14 +144,51 @@ export const XRPLTransaction: FC<XRPLTxProps> = ({
...
+    const formatted = value.issuer ? `${currency}\n${value.issuer}` : currency;
```
Comment (FlorianBouron) [inline packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx:151]: currency is used only one time, so no need to put in a variable:
```suggestion
    const formatted = value.issuer ? `${currency}\n${value.issuer}` : convertHexCurrencyString(value.currency);
```

File: packages/extension/src/utils/format.ts [hunk 6]
```diff
@@ -121,6 +127,15 @@ const LABEL_SET_NO_RIPPLE = 'Set No Ripple';
 const LABEL_CLEAR_NO_RIPPLE = 'Clear No Ripple';
 const LABEL_SET_FREEZE = 'Set Freeze';
 const LABEL_CLEAR_FREEZE = 'Clear Freeze';
+// AMMDeposit
+const LABEL_LPTOKEN = 'LP Token';
```
Comment (FlorianBouron) [inline packages/extension/src/utils/format.ts:131]: This is already in another file, maybe we should have only one source of truth

Comment (ThibautBremand) [reply packages/extension/src/utils/format.ts:131]: It's not the same purpose ! transaction flag VS Token listing

File: packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx (hunk 5 above)
Comment (ThibautBremand) [reply packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx:151]: currency appears twice?

Comment (FlorianBouron) [reply packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx:150]: Oh right!

**Review by FlorianBouron — APPROVED** [review]

**Pull request merged**